packages/web-components/src/components/va-file-input/va-file-input-upgrader.ts

```diff
@@ -364,12 +364,12 @@ const addPreviewHeading = (fileInputEl, fileNames) => {
  * @param {HTMLDivElement} dropTarget - The drag and drop target area.
  */
 
-const handleChange = (e, fileInputEl, instructions, dropTarget) => {
+const handleChange = (e: Event, fileInputEl: HTMLInputElement, instructions: HTMLDivElement, dropTarget: HTMLDivElement) => {
   const multiple = fileInputEl.getAttribute('multiple'); // TODO: Double-check this works correctly when multiple is re-enabled
-  const fileNames = multiple ? e.target.files : [e.target.files[0]];
+  const fileNames = multiple ? (e.target as HTMLInputElement).files : [(e.target as HTMLInputElement).files[0]];
```

Comment:
Instead of casting the type twice, you could store `e.target` as a separate variable and cast it once maybe?